### ISSUE TITLE
Pipeline in PR Close

### ISSUE THREAD

**Issue by jmendiara** [buildkite/feedback#433]
Use Case:

1. Devs create the Pull Request with number 512
2. CI launches a Pull Request pipeline[1]
3. A new deployment is created, for example (https://pr512.dev.example.com)
4. Some automated e2e testing is done
5. Devs connect to the environment and make manual testing (very common when developing UI)
6. Devs pushes more commits. Goto 2.
7. Changeset is accepted and the PR is merged
8. https://pr512.dev.example.com is undeployed

To make this happen you need to listen to github webhooks by your own and trigger pipelines via api.

It would be nice to have buildkite native support for this.

[1] Pull Request pipeline understood as something only runs in PR, configured this way
<img width="878" alt="captura de pantalla 2018-07-21 a las 8 33 56" src="https://user-images.githubusercontent.com/851359/43033150-57d47f36-8cc5-11e8-916f-a73e820630ba.png">

**Comment by petemounce** [comment]
I'm not sure I've parsed this correctly, but I think you're asking for the ability to run a pipeline, or steps, when a PR is merged. I'd definitely find that useful - I'd use it to 
* clean up remote branches
* reap now-unnecessary cloud resources to save costs

at the least.

**Comment by jmendiara** [comment]
Yep, not only on PR merge, but PR close (that is, when merged or discarded)

**Comment by jdotjdot** [comment]
We'd use this too.  Currently we make this happen by having another microservice receive Github webhooks, watch out for PR merges, and then manually call a Buildkite pipeline to discard old review environments. I'd rather it be part of Buildkite directly, obviously.

**Comment by miles-** [comment]
This is something that would be useful for us as well. We are planning to implement this with a cloud function that listens for pull request events and triggers the pipeline API for close events, but it'd be nicer to have this built directly into Buildkite.

**Comment by williamcodes** [comment]
I'd like this feature for deleting branch preview environments. IS this still a live issue?

**Comment by zeele** [comment]
+1

**Comment by bluemalkin** [comment]
Any update on this ? I am forced to use github actions to call the BK api to trigger an undeploy pipeline.

**Comment by s33dunda** [comment]
Yes please. Don't force me into the arms of GHA. :D

**Comment by andrewdrob10** [comment]
Just wondering how this is not a feature already?

As a workaround, we are using GitHub Actions to trigger a BuildKite pipeline to clean up our cloud infrastructure that is deployed using BuildKite? Seems a bit of a backwards approach to cleaning up on a `closed` event. Is there anything on the roadmap for this?
